static/app/components/profiling/flamegraph.tsx

```diff
@@ -78,8 +76,10 @@ function Flamegraph(props: FlamegraphProps): ReactElement {
         />
         <ThreadMenuSelector
           profileGroup={props.profiles}
-          activeProfileIndex={flamegraph.profileIndex}
-          onProfileIndexChange={setActiveProfileIndex}
+          activeProfileIndex={activeProfileIndex ?? 0}
```

Comment:
If you make the above change, you should be able to go back to using `flamegraph.profileIndex`